### PR TITLE
Term entry] java array list .sort

### PR DESCRIPTION
**Pull request by Brag2gr8**
### Description

- Added the Calendar .sort() term for Java
- Closing issue #4102 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/java/concepts/array-list/terms/sort/sort.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+```pseudo
+Arrays.sort(array);
+```
+
+## Examples
+
+### Example 1
+
+In this example, the `Arrays.sort(numbers)` method is used to sort the array of integers in ascending order. The result is then printed using `Arrays.toString()`.
```
Comment (dakshdeepHERE) [inline content/java/concepts/array-list/terms/sort/sort.md:29]:
```suggestion
In this example, the `Arrays.sort(numbers)` method is used to sort the array of integers in ascending order. The result is then printed using `Arrays.toString()`:
```

File: content/java/concepts/array-list/terms/sort/sort.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+}
+```
+
+The above example will give the following output:
+
+```shell
+Sorted Array: [1, 2, 3, 5, 8]
+```
+
+> **Note:** For an array of objects, you can use a custom comparator or ensure that the objects in the array implement the `Comparable` interface. Here's an example with strings.
```
Comment (dakshdeepHERE) [inline content/java/concepts/array-list/terms/sort/sort.md:52]: Avoid using first- and second-person pronouns (e.g. "I", "we", "you") if possible.

File: content/java/concepts/array-list/terms/sort/sort.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+### Example 2
+
+In this Example, strings implement the `Comparable` interface, so they can be sorted directly using Arrays.sort().
```
Comment (dakshdeepHERE) [inline content/java/concepts/array-list/terms/sort/sort.md:56]:
```suggestion
In this example, strings implement the `Comparable` interface, so they can be sorted directly using `Arrays.sort()`:
```

File: content/java/concepts/array-list/terms/sort/sort.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+```shell
+Sorted Array: [Alice, Bob, Charlie, David]
+```
+
+> **Note:** For a custom sorting order, the `Comparator` interface can be used. Here's an example sorting strings by their length.
+
+### Example 3
+
+In this example, `Comparator.comparing(String::length)` is used to sort the array of strings based on their length.
```
Comment (dakshdeepHERE) [inline content/java/concepts/array-list/terms/sort/sort.md:84]:
```suggestion
In this example, `Comparator.comparing(String::length)` is used to sort the array of strings based on their length:
```

File: content/java/concepts/array-list/terms/sort/sort.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    String[] names = {"Alice", "Bob", "Charlie", "David"};
+
+    // Sorting the array of strings by length
+    Arrays.sort(names, Comparator.comparing(String::length));
+
+    System.out.println("Sorted Array by Length: " + Arrays.toString(names));
+  }
+}
+```
+
```
Comment (dakshdeepHERE) [inline content/java/concepts/array-list/terms/sort/sort.md:101]:
```suggestion

The above example will give the following output:

```

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
I have requested few changes from my side.

**Comment by Brag2gr8** [comment]
@dakshdeepHERE done commiting the changes

**Review by dakshdeepHERE — APPROVED** [review]
LGTM 🚀

File: content/java/concepts/array-list/terms/sort/sort.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+---
+
+The **`.sort()`** method is used to sort arrays of primitive types and objects. The sorting is done in ascending order by default, but it can be customized for objects by implementing the `Comparable` interface or using a `Comparator`.
+
+## Syntax
+
+Here's a basic syntax of how to use the `.sort()` method:
+
+```pseudo
+Arrays.sort(array);
```
Comment (avdhoottt) [inline content/java/concepts/array-list/terms/sort/sort.md:22]: As you've added in the introduction of the method, there can be an optional parameter `Comparator`. Please add it to the syntax and write a one-line description below the syntax block.

File: content/java/concepts/array-list/terms/sort/sort.md [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+import java.util.Arrays;
+import java.util.Comparator;
+
+public class Main {
+  public static void main(String[] args) {
+    String[] names = {"Alice", "Bob", "Charlie", "David"};
+
+    // Sorting the array of strings by length
+    Arrays.sort(names, Comparator.comparing(String::length));
+
```
Comment (avdhoottt) [inline content/java/concepts/array-list/terms/sort/sort.md:96]:
```suggestion
```

File: content/java/concepts/array-list/terms/sort/sort.md [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+```java
+import java.util.Arrays;
+
+public class Main {
+  public static void main(String[] args) {
+    int[] numbers = {5, 2, 8, 1, 3};
+
+    // Sorting the array in ascending order
+    Arrays.sort(numbers);
+
```
Comment (avdhoottt) [inline content/java/concepts/array-list/terms/sort/sort.md:40]:
```suggestion
```

File: content/java/concepts/array-list/terms/sort/sort.md [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+}
+```
+
+The above example will give the following output:
+
+```shell
+Sorted Array: [Alice, Bob, Charlie, David]
+```
+
+> **Note:** For a custom sorting order, the `Comparator` interface can be used. Here's an example sorting strings by their length.
```
Comment (avdhoottt) [inline content/java/concepts/array-list/terms/sort/sort.md:80]:
```suggestion
> **Note:** For a custom sorting order, the `Comparator` interface can be used. Here's an example of sorting strings by their length.
```

**Review by avdhoottt — CHANGES_REQUESTED** [review]
@Brag2gr8 Few changes from my side.

**Comment by Brag2gr8** [comment]
@avdhoottt done with the changes

File: content/java/concepts/array-list/terms/sort/sort.md [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+}
+```
+
+The above example will give the following output:
+
+```shell
+Sorted Array: [1, 2, 3, 5, 8]
+```
+
+> **Note:** For an array of objects, a custom comparator can be used or the objects in the array should implement the `Comparable` interface. Here's an example with strings.
```
Comment (avdhoottt) [inline content/java/concepts/array-list/terms/sort/sort.md:53]:
```suggestion
> **Note:** For an array of objects, a custom comparator can be used or the objects in the array should implement the `Comparable` interface.
```

**Review by avdhoottt — APPROVED** [review]
Let's get this merged.

**Pull request merged**